src/instructionFiles/InstructionFileManager.ts

```diff
@@ -0,0 +1,311 @@
+  };
+
+  private readonly dialogHandler: DialogHandler;
+  private readonly editorTypeProvider: () => EditorType;
+
+  constructor(
+    dialogHandler: DialogHandler = new VsCodeDialogHandler(),
+    editorTypeProvider?: () => EditorType
+  ) {
+    this.dialogHandler = dialogHandler;
+    this.editorTypeProvider = editorTypeProvider || (() => this.detectEditorType());
+  }
+
+  private detectEditorType(): EditorType {
+    const appName = vscode.env.appName || '';
+
+    if (appName === 'Visual Studio Code') {
+      return EditorType.VSCode;
+    } else if (appName === 'Cursor') {
+      return EditorType.Cursor;
+    } else {
+      return EditorType.VSCode;
+    }
+  }
+
+  private getPromptMode(): string {
+    const config = vscode.workspace.getConfiguration('NPL');
+    return config.get<string>('instructionPrompts.mode', this.PROMPT_MODES.ASK);
+  }
+
+  private async setPromptMode(mode: string): Promise<void> {
+    const config = vscode.workspace.getConfiguration('NPL');
+    await config.update('instructionPrompts.mode', mode, vscode.ConfigurationTarget.Global);
+  }
+
+  public async checkAndHandleInstructionFiles(workspaceFolder: vscode.WorkspaceFolder): Promise<void> {
+    if (!workspaceFolder) {
+      return;
+    }
+
+    const promptMode = this.getPromptMode();
+    if (promptMode === this.PROMPT_MODES.DISABLED) {
+      return;
+    }
+
+    const editorType = this.editorTypeProvider();
+    const filePath = editorType === EditorType.VSCode ?
+      COPILOT_INSTRUCTIONS_PATH :
+      CURSOR_RULES_PATH;
+
+    await this.checkAndHandleInstructionFile(workspaceFolder, filePath, promptMode);
+  }
+
+  public async checkAndHandleCopilotInstructions(workspaceFolder: vscode.WorkspaceFolder): Promise<void> {
+    const promptMode = this.getPromptMode();
+    if (promptMode !== this.PROMPT_MODES.DISABLED) {
+      await this.checkAndHandleInstructionFile(workspaceFolder, COPILOT_INSTRUCTIONS_PATH, promptMode);
+    }
+  }
+
+  public async checkAndHandleCursorRules(workspaceFolder: vscode.WorkspaceFolder): Promise<void> {
+    const promptMode = this.getPromptMode();
+    if (promptMode !== this.PROMPT_MODES.DISABLED) {
+      await this.checkAndHandleInstructionFile(workspaceFolder, CURSOR_RULES_PATH, promptMode);
+    }
+  }
+
+  private async checkAndHandleInstructionFile(
+    workspaceFolder: vscode.WorkspaceFolder,
+    filePath: string,
+    promptMode: string
+  ): Promise<void> {
+    const fullPath = path.join(workspaceFolder.uri.fsPath, filePath);
+    const isAutoMode = promptMode === this.PROMPT_MODES.AUTO;
+
+    if (!fs.existsSync(fullPath)) {
+      if (isAutoMode) {
+        // Auto mode - create without asking
+        await this.createInstructionFile(fullPath, this.templatePath);
+        return;
+      }
+
+      // Ask mode - prompt the user
+      const answer = await this.dialogHandler.showInformationMessage(
+        'Create and maintain AI rules file with NPL-specific instructions?',
```

Comment:
I made these intentionally generic (no mention of which editor and kind of file or path) to keep the code simple. I also think it happens to be more (noob)-user-friendly.